transaction/std/src/memo_builder/gift_code_cancellation_memo_builder.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Defines the Memo Builder for the gift code cancellation memo (0x0202)
+//! specified in MCIP #32
+
+use super::{
+    memo::{GiftCodeCancellationMemo, UnusedMemo},
+    MemoBuilder, ReservedDestination,
+};
+use mc_account_keys::PublicAddress;
+use mc_transaction_core::{Amount, MemoContext, MemoPayload, NewMemoError};
+
+/// There are three possible gift code memo types specified in MCIP #32
+/// | Memo type bytes | Name                                              |
+/// | -----------     | -----------                                       |
+/// |    0x0002       | Gift Code Sender Memo                             |
+/// |    0x0201       | Gift Code Funding Memo                            |
+/// | -->0x0202<--    | Gift Code Cancellation Memo                       |
```

Comment:
Nit: The table does not need to be so wide
```suggestion
/// | Memo type bytes | Name                          |
/// | -----------     | -----------                   |
/// |    0x0002       | Gift Code Sender Memo         |
/// |    0x0201       | Gift Code Funding Memo        |
/// | -->0x0202<--    | Gift Code Cancellation Memo   |
```